ngraph/core/include/ngraph/runtime/reference/mvn.hpp

```diff
@@ -0,0 +1,78 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cstddef>
+#include <ngraph/runtime/opt_kernel/broadcast.hpp>
+#include <ngraph/runtime/reference/mean.hpp>
+#include <ngraph/runtime/reference/multiply.hpp>
+#include <ngraph/runtime/reference/sqrt.hpp>
+#include <ngraph/runtime/reference/subtract.hpp>
+#include <ngraph/runtime/reference/sum.hpp>
+#include <ngraph/shape.hpp>
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            template <typename T>
+            void mvn(const T* arg,
+                     T* out,
+                     const Shape& in_shape,
+                     bool normalize_variance,
+                     AxisSet reduction_axes,
+                     double eps)
+            {
+                auto reduced_shape = reduce(in_shape, reduction_axes, true);
+                std::vector<T> mean_val(shape_size(reduced_shape));
+                mean(arg, mean_val.data(), in_shape, reduction_axes, true);
+                std::vector<T> broadcast_mean_data(shape_size(in_shape));
```

Comment:
If we can get rid of broadcasts, I guess we can also get rid of the intermediate buffers. I guess it should be safe to use the same buffer as source and destination in case of arithmetic operations.  I'll try

Reply:
Well, but looks like it doesn't safe. I tried to get rid of one intermediate buffer like bellow and got an accuracy mismatch
```cpp
multiply(out, out, out, shape_size(in_shape));
std::vector<T> sum_val(shape_size(reduced_shape));
sum(out, sum_val.data(), in_shape, reduction_axes, true);
```